internal/v2/s2av2_e2e_test.go

```diff
@@ -128,9 +132,33 @@ func startServer(t *testing.T, s2aAddress string, localIdentities []*commonpbv1.
 	return lis.Addr().String()
 }
 
+// startFallbackServer runs a GRPC echo testing server.
+// It's used to test the default fallback logic upon S2A failure.
+func startFallbackServer(t *testing.T) string {
+	lis, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Errorf("net.Listen(tcp, :0) failed: %v", err)
+	}
+	cert, err := tls.X509KeyPair(serverCertpem, serverKeypem)
+	if err != nil {
+		t.Errorf("failure initializing tls.certificate: %v", err)
+	}
+	creds := credentials.NewTLS(&tls.Config{
```

Comment:
Please add a comment that explains that we intentionally are not requesting client certs. :)